src/Ilios/CoreBundle/Traits/CalendarEventRepository.php

```diff
@@ -0,0 +1,359 @@
+<?php
+
+namespace Ilios\CoreBundle\Traits;
+
+use Doctrine\ORM\EntityManager;
+use Ilios\CoreBundle\Classes\CalendarEvent;
+use Ilios\CoreBundle\Classes\UserMaterial;
+use Ilios\CoreBundle\Service\UserMaterialFactory;
+
+/**
+ * Class CalendarEventRepository
+ *
+ * Loads CalendarEvents
+ */
+trait CalendarEventRepository
+{
+    /**
+     * Convert offerings into CalendarEvent() objects
+     * @param integer $userId
+     * @param array $results
+     *
+     * @return CalendarEvent[]
+     */
+    protected function createEventObjectsForOfferings($userId, array $results)
+    {
+        return array_map(function ($arr) use ($userId) {
+            $event = new CalendarEvent();
+            $event->user = $userId;
```

Comment:
`$user` is a property of `UserEvent`, not `CalendarEvent`. suggested change is to cast calendar events to user events further downstream and set the user id then (similar to casting to school ids).